### PR TITLE
Add logdensity_gradient_and_hessian.

### LINKED ISSUE #65: Feature request: `logdensity_and_hessian`
The `logdensity_and_gradient` function is very useful for using e.g. `ForwardDiff` to compute the gradient of the log density function.

Could we also add a new function `logdensity_and_hessian` that uses `ForwardDiff` to compute the Hessian of the log density function?

Although, I suppose that, in the process of computing the Hessian, `ForwardDiff` will also compute and store the gradient. So it would probably make the most sense to add a new function `logdensity_and_gradient_and_hessian` that returns the value, the gradient, and the Hessian.

### PR DESCRIPTION
**Pull request by tpapp**
Fixes #65.

### CODE REVIEW

File: src/LogDensityProblems.jl [hunk 1]
```diff
@@ -124,6 +127,31 @@ The first argument (the log density) can be shifted by a constant, see the note
...
+has length compatible with its [`dimension`](@ref).
+
+Return three values:
+
+- the log density as real number, which equivalent to `logdensity(ℓ, x)`
+
+- *if* the log density is finite, the gradient, an `::AbstractVector` of real numbers,
+   otherwise this value is arbitrary and should be ignored.
+
+- *if* the log density is finite, the Hessian, an `::AbstractMatrix` of real numbers,
```
Comment (devmotion) [inline src/LogDensityProblems.jl:143]: Should the interface include a note about non-finite gradient (elements) as well?

File: src/utilities.jl [hunk 2]
```diff
@@ -37,8 +37,9 @@ Test `ℓ` with random values.
 `N` random vectors are drawn from a standard multivariate Cauchy distribution, scaled with
 `scale` (which can be a scalar or a conformable vector).
 
-Each random vector is then used as an argument in `f(ℓ, ...)`. `logdensity` and
-`logdensity_and_gradient` are  recommended for `f`.
+Each random vector is then used as an argument in `f(ℓ, ...)`. [`logdensity`](@ref),
```
Comment (devmotion) [inline src/utilities.jl:40]: Maybe more explicit:
```suggestion
Each random vector `x` is then used as an argument in `f(ℓ, x)`. [`logdensity`](@ref),
```

File: src/LogDensityProblems.jl (hunk 1 above)
Comment (tpapp) [reply src/LogDensityProblems.jl:143]: My intention for the API is the implicit assumption that if the log density is finite, so are all derivatives, but this is not explicitly documented.

The motivation for the above is that the caller only needs to check the log density to make sure things make sense. I will open a separate issue for this though as it goes beyond this PR.

Comment (tpapp) [reply src/LogDensityProblems.jl:143]: Cf #102.

**Review by devmotion — APPROVED** [review]

**Pull request merged**